src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -121,9 +207,38 @@ BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, IN_MEMORY_READ)(benchmark::S
   }
 }
 
+BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, IN_MEMORY_READ_RANDOM)(benchmark::State& state) {  // open file
+  for (auto _ : state) {
+    const int32_t NUMBER_OF_BYTES = state.range(0) * MB;
+
+    state.PauseTiming();
+    std::vector<uint32_t> read_data;
+    auto random_read_amount = static_cast<size_t>(NUMBER_OF_BYTES / 4);
+    read_data.resize(random_read_amount);
+    state.ResumeTiming();
+
+    for (auto index = size_t{0}; index < random_read_amount; ++index) {
+      read_data[index] = numbers[random_indices[index]];
```

Comment:
This way the read vector is in a different order than the vector saved in memory.

Reply:
Lets ask in the meeting if this is a problem